src/QueryFilterHelpers.ts

```diff
@@ -0,0 +1,604 @@
+  if (equal.operand[0].type == 'FunctionRef') {
+    propRef = interpretFunctionRef(equal.operand[0] as ELMFunctionRef);
+  } else if (equal.operand[0].type == 'Property') {
+    propRef = equal.operand[0] as ELMProperty;
+  }
+
+  if (propRef == null) {
+    console.warn('could not resolve property ref for Equivalent');
+    return { type: 'unknown' };
+  }
+
+  if (equal.operand[1].type == 'Literal') {
+    const literal = equal.operand[1] as ELMLiteral;
+    if (propRef.scope && literal.value) {
+      return {
+        type: 'equals',
+        alias: propRef.scope,
+        value: literal.value,
+        attribute: propRef.path,
+        localId: equal.localId
+      };
+    } else {
+      console.warn('Property reference scope or literal value were not found.');
+      return { type: 'unknown' };
+    }
+  }
+
+  if (equal.operand[1].type == 'ConceptRef') {
+    const conceptRef = equal.operand[1] as ELMConceptRef;
+    if (propRef.scope) {
+      return {
+        type: 'in',
+        alias: propRef.scope,
+        attribute: propRef.path,
+        valueCodingList: getCodesInConcept(conceptRef.name, library),
+        localId: equal.localId
+      };
+    } else {
+      console.warn('Property reference scope was not found.');
+      return { type: 'unknown' };
+    }
+  }
+  return { type: 'unknown' };
+}
+
+/**
+ * Gets a list of codes in a CQL concept reference by name. These are returned as
+ * FHIR Codings.
+ *
+ * @param name Name of the concept.
+ * @param library The library elm the concept should be found in.
+ * @returns A list of codings in the concept
+ */
+function getCodesInConcept(name: string, library: ELM): R4.ICoding[] {
+  const concept = library.library.concepts?.def.find(concept => concept.name === name);
+  if (concept) {
+    const codes: R4.ICoding[] = [];
+    concept.code.map(codeRef => {
+      const code = library.library.codes?.def.find(code => code.name == codeRef.name);
+      if (code) {
+        codes.push({
+          code: code?.id,
+          system: library.library.codeSystems.def.find((systemRef: any) => code?.codeSystem.name === systemRef.name).id
+        });
+      }
+    });
+    return codes;
+  }
+  return [];
+}
+
+/**
+ * Parses an ELM equivalent expression into a filter. This currently only handles checks against literal values.
+ *
+ * @param equal The equal expression to be parsed.
+ * @returns Filter representing the equal filter.
+ */
+function interpretEqual(equal: ELMEqual): EqualsFilter | UnknownFilter {
+  let propRef: ELMProperty | null = null;
+  if (equal.operand[0].type == 'FunctionRef') {
+    propRef = interpretFunctionRef(equal.operand[0] as ELMFunctionRef);
+  } else if (equal.operand[0].type == 'Property') {
+    propRef = equal.operand[0] as ELMProperty;
+  }
+
+  let literal: ELMLiteral | null = null;
+  if (equal.operand[1].type == 'Literal') {
+    literal = equal.operand[1] as ELMLiteral;
+  }
+
+  if (propRef?.scope && literal?.value) {
```

Comment:
Thinking from a "how are we going to take this information and plop it onto GuidanceResponse" perspective, do you have any insight on how we'd aim to distinguish Equals expressions that belong in `codeFilter` vs something else like a valueFilter extension?

I know this is kind of beyond the scope of the blind query parsing here, but I'm trying to gauge if there's any more information in the ELM we can gather to indicate the differences here. Otherwise it may be tough on the modeling end to take these equals expressions and smartly put them on a GuidanceResponse dataRequirement.

Reply:
This is a difficult one. If it's checking if a FHIR `code` (read string) is equal to a CQL string ex. `Procedure.status = 'complete'`... then we really have no clue if the attribute being checked is a code or not from the ELM. I think we possibly would need a lookup table to determine what type that attribute is.

Reply:
That's what I figured. Something to think about for the future, but I'm good as is for now!